django/db/models/fields/related.py

```diff
@@ -697,18 +697,17 @@ def get_extra_restriction(self, where_class, alias, related_alias):
         """
         return None
 
-    def get_path_info(self):
+    def get_path_info(self, filtered_relation=None):
         """Get path from this field to the related model."""
         opts = self.remote_field.model._meta
         from_opts = self.model._meta
-        return [PathInfo(from_opts, opts, self.foreign_related_fields, self, False, True)]
+        return [PathInfo(from_opts, opts, self.foreign_related_fields, self, False, True, filtered_relation)]
```

Comment:
You turned the arguments into keyword arguments in one other place. Could you also do this here and below? I don't know if it makes sense to also do this in the tests.